### PR TITLE
[Search v2.4] Update Search results page (Search bar only)

### PR DESCRIPTION
**Pull request by cdOut**
### Details

Reuses SearchRouterInput inside Search results page to display the result search query with a filters button inside itself (it is currently disabled as autocomplete and support for typed queries will be part of Search 2.5).

### Fixed Issues

$ https://github.com/Expensify/App/issues/49121
PROPOSAL:

### Tests

1. Log into an account that preferably has data to be searched / create some expenses
2. Go into the Search tab
3. Using the filters button create any search query you want and confirm it with `View results`
4. Confirm that the entire search query in a string format displays properly in the top bar of search results page, with a filters button to the right of itself which is accessible to modify the query.

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/6bead7a2-cda3-416b-84ba-ed302203c677

</details>

### CODE REVIEW

**Comment by luacmartins** [comment]
I'm really excited for this one. It's looking good so far!

File: src/components/Search/SearchPageHeader.tsx
Comment (Kicu) [inline src/components/Search/SearchPageHeader.tsx:39]: I'd rename this something more simple like `text` or `message` or similar. 
Content suggests to me that this is a more complex structure, and at first I was confused why we have both `children` and `content`

**Review by Kicu — APPROVED** [review]
Looks mostly good 👍 I left several small comment, please fix them and it's ready to go

File: src/components/TextInput/BaseTextInput/index.tsx
Comment (cdOut) [reply src/components/TextInput/BaseTextInput/index.tsx:69]: I agree that this isn't perfect but there wasn't any other way to disable the default styling when disabling the input. I also second the simplification of `BaseTextInput`.

**Comment by luacmartins** [comment]
>(it is currently disabled as autocomplete and support for typed queries will be part of Search 2.5)

@cdOut autocomplete is a 2.5 feature, but typed queries is part of the scope of v2.4. @Kicu already introduced that functionality [here](https://github.com/Expensify/App/pull/48785), so we need to enable the input on this page.


https://github.com/user-attachments/assets/209cfb4d-f2ed-4403-8cd1-415c38a89351

**Review by luacmartins — COMMENTED** [review]
>(it is currently disabled as autocomplete and support for typed queries will be part of Search 2.5)

@cdOut autocomplete is a 2.5 feature, but typed queries is part of the scope of v2.4. @Kicu already introduced that functionality [here](https://github.com/Expensify/App/pull/48785), so we need to enable the input on this page. We can handle enabling the input as a follow up if you'd like to keep it separate


https://github.com/user-attachments/assets/209cfb4d-f2ed-4403-8cd1-415c38a89351

**Review by rayane-d — CHANGES_REQUESTED** [review]
Looks good! I've left a few minor comments

**Comment by Kicu** [comment]
@luacmartins @cdOut 
About this comment:

> autocomplete is a 2.5 feature, but typed queries is part of the scope of v2.4. @Kicu already introduced that functionality https://github.com/Expensify/App/pull/48785, so we need to enable the input on this page. We can handle enabling the input as a follow up if you'd like to keep it separate

I think doing this in a follow up PR is the way to go.
Right now we can create multiple complex queries using `AdvancedFilters` including filters like from:, to:, tag: etc. 
When this input becomes editable, the query produced with values from advanced filters has to be made editable and work together with manually typed query. We're expecting that some weird bugs and edge cases might start to appear then, so it's better to turn it on separately and test throroughly.

**Comment by dannymcclain** [comment]
@rayane-djouah What color is being used for the background of the input? In Figma, we have it set to `row-hover`. In your screenshot it looks like maybe it's using `highligh-background`?

Also, for the text inside the input, if possible, we have it laid out in Figma where the name of the filter uses `text-supporting` and the value of the filter uses our standard `text` color. Is that something we can accomplish here? (so for example, **type:** would be set in `text-supporting` and **expense** would be set in `text`)

cc @dubielzyk-expensify for a gut check on all that—but that's what I'm seeing in Figma.

Also last little thing: can we make it to where when the query gets "cut off" or truncated by the field, we add a little `...` to the end like we do elsewhere in the app?

**Comment by dannymcclain** [comment]
>Yes, it's using highlightBG, do we need to use theme.sidebarHover (colors.productLight300 and colors.productDark300)?

I'm not sure what the variable is called in the code. We call it `row-hover` in Figma and it maps to this hex code: `#0A2E25`

>I'm not sure if this is within the scope of this PR since it involves markdown formatting

Totally fair, let's just leave it all as `text-supporting` for now.

>Change the text color of the "Filters" button to styles.textSupporting
>Change the font size from 13px to 15px
>Change the "Filters" icon size from 16*16px to 20*20px

I actually don't think we need to mess with this Filters button right now. I think we can leave the current colors/sizes of everything as is and just add in the hover state. I think for hover, all we need to do is update the background color. And also, I mentioned this in Figma, but can we add a tiny bit of space to the right of the filters button so that when it's hovered there's equal space around it on all sides? Looks like maybe right now it's about 8px away from the top and bottom, but it touches the edge of the input on the right.

>which hover background color should we use for the button?

We should be able to use our standard button hover color. We call it `button-hover` in Figma, but it translates to the hex code `#224F41`.

For the focused outline, we can totally do that as a follow up and wait until this field isn't disabled.

**Comment by dubielzyk-expensify** [comment]
Aligned to everything @dannymcclain is saying

> As demonstrated in https://github.com/Expensify/App/pull/49462#issuecomment-2367074261, we have enabled horizontal scrolling on the input field. This feature allows users to view and edit text that exceeds the boundaries of the input box. Therefore, adding an ellipsis ("...") at the end should be unnecessary, right?

Just clarifying that this doesn't show any sort of scroll bar on any platforms right? I agree that we shouldn't use ellipsis for the input field.

**Comment by cdOut** [comment]
I’ll be changing the designs today, will let you know once it’s all ready to be looked at again 👍

**Comment by 289Adam289** [comment]
Hi! I will slowly start to work on the follow up from this https://github.com/Expensify/App/pull/49462#issuecomment-2368051617. It is closely related to searchParser and headerTitle that i have worked on.

**Comment by dannymcclain** [comment]
>I agree that we shouldn't use ellipsis for the input field.

Oh that's my bad about the `...` in the input 🤦‍♂️ I didn't realize that was scrollable/wasn't thinking about it like an actual input. So yes I agree with you all that we should not include the ellipsis.

**Comment by luacmartins** [comment]
Thank you both @cdOut @289Adam289

File: src/components/Search/SearchRouter/SearchRouterInput.tsx
Comment (cdOut) [reply src/components/Search/SearchRouter/SearchRouterInput.tsx:1]: What browser are you using here?

Comment (Kicu) [reply src/components/Search/SearchRouter/SearchRouterInput.tsx:1]: FYI very soon in a separate PR we will make this input editable and not `disabled`. Can someone verify if this bug would also happen when the input is enabled? because if no then we could ignore this for now.

Comment (Kicu) [reply src/components/Search/SearchRouter/SearchRouterInput.tsx:1]: sigh, tested this myself and the same behavior is there if the input is not `disabled`. 
How big of a bug do we consider this? Scrolling on text behaves ok, perhaps that is enough?

Comment (luacmartins) [reply src/components/Search/SearchRouter/SearchRouterInput.tsx:1]: I don't think this is a blocker for this PR. We can definitely address it as a follow up.

**Comment by Kicu** [comment]
@cdOut you have a TS error

File: src/components/Search/SearchRouter/SearchRouterInput.tsx
Comment (cdOut) [reply src/components/Search/SearchRouter/SearchRouterInput.tsx]: Should be all fixed, none of those occured anymore when testing on chrome on the latest commit.

**Comment by cdOut** [comment]
@rayane-djouah ready for a re-review, also merged with the latest main

**Review by rayane-d — CHANGES_REQUESTED** [review]

File: src/components/Search/SearchRouter/SearchRouterInput.tsx
Comment (cdOut) [reply src/components/Search/SearchRouter/SearchRouterInput.tsx:62]: I would prefer leaving it this way, since using `paddingHorizontal` here gets overwritten by internal styling of TextInput, leaving only the right padding intact and nullifying the left one. When specifying the paddings directly they get presented correctly.

**Comment by cdOut** [comment]
@rayane-djouah I've added the changes and also specified why I'd leave one of the parts of styling as is, let me know if anything else needs fixing before approve

**Comment by cdOut** [comment]
Resolved merge conflicts with `SearchPageHeader`

**Review by rayane-d — APPROVED** [review]
LGTM and tests well

**Review by luacmartins — DISMISSED** [review]
LGTM

**Comment by luacmartins** [comment]
NAB we might wanna add some right padding to the button
Screenshot 2024-10-01 at 7 55 57 AM

**Comment by dannymcclain** [comment]
>NAB we might wanna add some right padding to the button

Big agree! I pointed this out somewhere else. But ideally we'd have equal padding on all sides. Definitely NAB though.

**Comment by cdOut** [comment]
> NAB we might wanna add some right padding to the button Screenshot 2024-10-01 at 7 55 57 AM

Ah that’s right, I can add it in a moment if we can hold off on merging it right away.

**Review by luacmartins — APPROVED** [review]

File: src/components/Search/SearchPageHeader.tsx
Comment (eh2077) [inline src/components/Search/SearchPageHeader.tsx:332]: Setting `pressOnEnter` prop here caused a regression https://github.com/Expensify/App/issues/51427

**Pull request merged**
